extensions/gitpod-web/src/ports/index.ts

```diff
@@ -0,0 +1,284 @@
+		});
+		return port.setPortVisibility('private');
+	}));
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.ports.makePublic', (port: GitpodWorkspacePort) => {
+		context.fireAnalyticsEvent({
+			eventName: 'vscode_execute_command_gitpod_ports',
+			properties: { action: 'public' }
+		});
+		return port.setPortVisibility('public');
+	}));
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.ports.tunnelNetwork', (port: GitpodWorkspacePort) => {
+		port.setTunnelVisibility(TunnelVisiblity.NETWORK);
+	}));
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.ports.tunnelHost', async (port: GitpodWorkspacePort) =>
+		port.setTunnelVisibility(TunnelVisiblity.HOST)
+	));
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.ports.preview', (port: GitpodWorkspacePort) => {
+		context.fireAnalyticsEvent({
+			eventName: 'vscode_execute_command_gitpod_ports',
+			properties: { action: 'preview' }
+		});
+		return openPreview(port);
+	}));
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.ports.openBrowser', (port: GitpodWorkspacePort) => {
+		context.fireAnalyticsEvent({
+			eventName: 'vscode_execute_command_gitpod_ports',
+			properties: { action: 'openBrowser' }
+		});
+		return port.openExternal();
+	}));
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.ports.retryAutoExpose', async (port: GitpodWorkspacePort) => {
+		const request = new RetryAutoExposeRequest();
+		request.setPort(port.portNumber);
+		await util.promisify(context.supervisor.port.retryAutoExpose.bind(context.supervisor.port, request, context.supervisor.metadata, {
+			deadline: Date.now() + context.supervisor.deadlines.normal
+		}))();
+	}));
+
+	const portsStatusBarItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right);
+	context.subscriptions.push(portsStatusBarItem);
+	function updateStatusBar(): void {
+		const exposedPorts: number[] = [];
+
+		for (const port of portViewProvider.portMap.values()) {
+			if (isExposedServedGitpodWorkspacePort(port)) {
+				exposedPorts.push(port.status.localPort);
+			}
+		}
+		let text: string;
+		let tooltip = 'Click to open "Ports View"';
+		if (exposedPorts.length) {
+			text = 'Ports:';
+			tooltip += '\n\nPorts';
+			text += ` ${exposedPorts.join(', ')}`;
+			tooltip += `\nPublic: ${exposedPorts.join(', ')}`;
+		} else {
+			text = '$(circle-slash) No open ports';
+		}
+
+		portsStatusBarItem.text = text;
+		portsStatusBarItem.tooltip = tooltip;
+		portsStatusBarItem.command = 'gitpod.ports.reveal';
+		portsStatusBarItem.show();
+	}
+	updateStatusBar();
+	context.subscriptions.push(portViewProvider.onDidChangePorts(() => updateStatusBar()));
+    // TODO(hw): What's this?
+	// context.subscriptions.push(vscode.commands.registerCommand('gitpod.ports.reveal', () => {
+	// 	workspaceView.reveal(gitpodWorkspaceTreeDataProvider.ports, {
+	// 		focus: true,
+	// 		expand: true
+	// 	});
+	// }));
```

Comment:
a command to show the ports view, it's used in the walktrough if you search for `gitpod.ports.reveal`